core/ledger/kvledger/txmgmt/statedb/stateleveldb/stateleveldb.go

```diff
@@ -268,3 +284,64 @@ func (scanner *kvScanner) GetBookmarkAndClose() string {
 	scanner.Close()
 	return retval
 }
+
+type fullDBScanner struct {
+	db               *leveldbhelper.DBHandle
+	dbItr            iterator.Iterator
+	skipNamespace    func(string) bool
+	currentNamespace string
+}
+
+func newFullDBScanner(db *leveldbhelper.DBHandle, skipNamespace func(string) bool) (*fullDBScanner, byte, error) {
+	dbItr := db.GetIterator(dataKeyPrefix, dataKeyStopper)
+	if err := dbItr.Error(); err != nil {
+		return nil, byte(0), errors.Wrap(err, "internal leveldb error while obtaining db iterator")
+	}
+	return &fullDBScanner{
+			db:            db,
+			dbItr:         dbItr,
+			skipNamespace: skipNamespace,
+		},
+		fullScanIteratorValueFormat,
```

Comment:
Not sure what you mean by that?

Reply:
Because the name is `fullScanIteratorValueFormat`, I thought we could have this variable as a member of `fullDBScanner`.

Reply:
This is too futuristic discussion - so, just for the sake of discussion - If you declare a variable inside there, it would mean that this supports multiple formats. Typically, if you have to support multiple formats, you would name the iterators like `format1Itr`, format2Itr.. and maintain a mapping between the formats and the iterators outside.